### PR TITLE
run rustfmt on librustc_metadata folder

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

File: src/librustc_metadata/encoder.rs [hunk 1]
```diff
@@ -1129,31 +1101,32 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
...
-        }).map(|filemap| &**filemap))
+        self.lazy_seq_ref(all_filemaps.iter()
+            .filter(|filemap| {
+                // No need to export empty filemaps, as they can't contain spans
+                // that need translation.
+                // Also no need to re-export imported filemaps, as any downstream
+                // crate will import them from their original source.
+                !filemap.lines.borrow().is_empty() && !filemap.is_imported()
+            })
+            .map(|filemap| &**filemap))
```
Comment (nrc) [inline src/librustc_metadata/encoder.rs]: I think this is somewhere we can teach Rustfmt to be a bit smarter, but its not the end of the world IMO. More annoying to me is that the alignment of dots is misleading about the scope.

**Comment by srinivasreddy** [comment]
@nrc  `approve` pls

File: src/librustc_metadata/encoder.rs [hunk 2]
```diff
@@ -1369,34 +1347,36 @@ pub fn encode_metadata<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
...
-        reexports: reexports,
-        link_meta: link_meta,
-        cstore: cstore,
-        reachable: reachable,
-        mir_map: mir_map,
-        lazy_state: LazyState::NoNode,
-        type_shorthands: Default::default(),
-        predicate_shorthands: Default::default()
-    }.encode_crate_root();
+            opaque: opaque::Encoder::new(&mut cursor),
```
Comment (jseyfried) [inline src/librustc_metadata/encoder.rs]: Strange indent (`rustfmt` bug?)

**Pull request merged**